### ISSUE TITLE
#error "SSE2 instruction set not enabled"

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1681]
**Reported by eggman on 4 Mar 2016 13:32 UTC**

root@Madhu:/src/aircrack-ng-1.2-rc4/src# make
gcc -g -W -Wall -O3  -pthread -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3   -fstack-protector -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude   -c -o sse-intrinsics.o sse-intrinsics.c
In file included from sse-intrinsics.c:85:0:
/usr/lib/gcc/i686-linux-gnu/4.7/include/emmintrin.h:32:3: error: #error "SSE2 instruction set not enabled"
make: *** [sse-intrinsics.o] Error 1

**Comment by aircrack-ng** [comment]
**Comment by misterx on 4 Mar 2016 13:32 UTC**

What is the content of common.cfg?

**Comment by aircrack-ng** [comment]
**Attachment by anonymous on 19 Mar 2016 18:46 UTC**

Uploaded file [1.JPG](https://github.com/aircrack-ng/trac-attachments/blob/master/1681/20160319-184624_450-1.JPG)

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 4 Mar 2016 13:32 UTC**

Getting same error. Common.cfg has nothing but a blank line in it. Snapshot attached.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 4 Mar 2016 13:32 UTC**

Could you paste lscpu?

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 4 Mar 2016 13:32 UTC**

I have the same issue when my cpu is old model pentium M, but it shows sse2 is supported.

```
[aircrack-ng-1.2-rc4](zchiml@localhost)$ lscpu
Architecture:          i686
CPU op-mode(s):        32-bit
Byte Order:            Little Endian
CPU(s):                1
On-line CPU(s) list:   0
Thread(s) per core:    1
Core(s) per socket:    1
Socket(s):             1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 13
Stepping:              6
CPU MHz:               600.000
BogoMIPS:              1196.92

[aircrack-ng-1.2-rc4](zchiml@localhost)$ cat /proc/cpuinfo
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 13
model name	: Intel(R) Pentium(R) M processor 2.00GHz
stepping	: 6
microcode	: 0x18
cpu MHz		: 600.000
cache size	: 2048 KB
fdiv_bug	: no
f00f_bug	: no
coma_bug	: no
fpu		: yes
fpu_exception	: yes
cpuid level	: 2
wp		: yes
flags		: fpu vme de pse tsc msr mce cx8 apic sep mtrr pge mca cmov clflush dts acpi mmx fxsr sse sse2 ss tm pbe bts est tm2
bogomips	: 1196.92
clflush size	: 64
cache_alignment	: 64
address sizes	: 32 bits physical, 32 bits virtual
power management:
```

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 4 Mar 2016 13:32 UTC**

I'm running CentOS 6 with gcc v4.4.7 (from official repositories) and clang v3.4.2 (from epel repository). Compiling with gcc produces the same error despite my CPU supporting both SSE and SSE2, but setting CC=clang produces a successful build with no errors regarding SSE2.


```
[~](root@Printers)# gcc --version
gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-18)
Copyright (C) 2010 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[~](root@Printers)# clang --version
clang version 3.4.2 (tags/RELEASE_34/dot2-final)
Target: i386-redhat-linux-gnu
Thread model: posix
[~](root@Printers)# lscpu
Architecture:          i686
CPU op-mode(s):        32-bit
Byte Order:            Little Endian
CPU(s):                1
On-line CPU(s) list:   0
Thread(s) per core:    1
Core(s) per socket:    1
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            15
Model:                 2
Model name:            Intel(R) Pentium(R) 4 CPU 2.53GHz
Stepping:              7
CPU MHz:               2532.954
BogoMIPS:              5065.90
NUMA node0 CPU(s):     0
[~](root@Printers)# cat /proc/cpuinfo
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 15
model           : 2
model name      : Intel(R) Pentium(R) 4 CPU 2.53GHz
stepping        : 7
microcode       : 0x37
cpu MHz         : 2532.954
cache size      : 512 KB
physical id     : 0
siblings        : 1
core id         : 0
cpu cores       : 1
apicid          : 0
initial apicid  : 0
fdiv_bug        : no
f00f_bug        : no
coma_bug        : no
fpu             : yes
fpu_exception   : yes
cpuid level     : 2
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe pebs bts cpuid cid xtpr
bugs            :
bogomips        : 5065.90
clflush size    : 64
cache_alignment : 128
address sizes   : 36 bits physical, 32 bits virtual
power management:
```

**Comment by aircrack-ng** [comment]
**Comment by misterx on 4 Mar 2016 13:32 UTC**

Should be fixed by aircrack-ng/aircrack-ng@4472d7dec98133e6f0df336457f2caf1c73d9842.
